src/docfx/build/xref/XrefResolver.cs

```diff
@@ -320,7 +318,15 @@ private static string RemoveSharingHost(string url, string hostName)
         if (_externalXrefMap.Value.TryGetValue(uid, out var spec))
         {
             var href = RemoveSharingHost(spec!.Href, _config.HostName);
-            return (spec, href);
+            if (_hostNameMap.ContainsKey(_config.HostName))
+            {
+                href = RemoveSharingHost(href, _hostNameMap[_config.HostName]);
+            }
+            if (_reverseHostNameMap.ContainsKey(_config.HostName))
+            {
+                href = RemoveSharingHost(href, _reverseHostNameMap[_config.HostName]);
+            }
+            return (spec, HostNameUtility.MigrateHostForUrl(href, _hostNameMap));
```

Comment:
Output legacy hostname before switch

Reply:
consider to add flag config to generate output with new hostname? 
**** 

MainHostName  ->  input + ouput 
AlternativeHostName -> input

****

Reply:
- hostname mapping
   - key: input + output 
   - value: input 
- reverse hostname mapping after switch